### PR TITLE
pkg/sysctl fixes: support commented KV lines, additional seperator and comment character, no post-crit KV set skips

### PR DESCRIPTION
**Pull request by eyz**
**- What I did**
- Correctly ignoring all commented KV lines, not only lines without equal signs
- Removing all whitespace in KVs (such as tabs as well) instead of only single spaces
- Added semicolon as a new comment prefix
- Added forward slash as separator character
- KV set fails are now logged as warnings with a prefix `WARN:`, and do not skip the rest of the KVs from being attempted (no silent fails/skips)

**- How I did it**
- Trimming the KV line and checking for hash (or semicolon) as first character and skipping line earlier in code block.
- Using `TrimSpace` to trim any whitespace, not only single space characters.
- Logging sysctl KV set error with `log.Println` instead of failing on first error with `log.Fatal`, so the rest of the KV sets can be attempted (instead of skipping them silently)

**- How to verify it**
- Lines such as the following now aren't attempted to be set: `#a.b.c=xyz`
- Lines such as the following are now skipped as comments `;a.b.c=xyz`
- Keys can be composed of periods `a.b.c=xyz`
- Keys can be composed of forward slashes `a/b/c=xyz`
- Disable IPv6 via cmdline `ipv6.disable=1` in kernel, and now `pkg/sysctl` won't skip the KV sets after `net.ipv6.conf.all.accept_redirects = 0`

**- Description for the changelog**

pkg/sysctl fixes: support commented KV lines, additional seperator and comment character, no post-crit KV set skips

**- A picture of a cute animal (not mandatory but encouraged)**
media-20171119

### CODE REVIEW

File: pkg/sysctl/main.go [hunk 1]
```diff
@@ -65,7 +68,7 @@ func main() {
 			}
 			err = sysctl(line)
 			if err != nil {
-				log.Fatal(err)
+				log.Println(fmt.Errorf("WARN: %v", err))
```
Comment (w9n) [inline pkg/sysctl/main.go]: I think this should be `log.Warnln(err.Error())` or `log.Fatalln(err.Error())`, otherwise LGTM to me.

Comment (eyz) [reply pkg/sysctl/main.go]: Thanks, @w9n! I didn't see Warnln

Comment (eyz) [reply pkg/sysctl/main.go]: Actually, @w9n, I just looked again - I don't see ant options other than `Println`, `Fatalln`, or `Panicln`, as this isn't using one of the third party logging libraries. See https://golang.org/pkg/log/

Comment (w9n) [reply pkg/sysctl/main.go]: sorry, i looked at `log "github.com/sirupsen/logrus"` in my project, so your Println seems good in the standard pkg

File: pkg/sysctl/build.yml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 image: sysctl
 config:
   pid: "host"
-  readonly: true
```
Comment (w9n) [inline pkg/sysctl/build.yml]: Why did you remove this one?

Comment (eyz) [reply pkg/sysctl/build.yml]: @w9n see the last line of each bullet above. I got errors about adding confs in via the image scoped YAML when read-only

File: pkg/sysctl/main.go (hunk 1 above)
Comment (eyz) [reply pkg/sysctl/main.go]: No problem. Thanks for the review!

File: pkg/sysctl/build.yml (hunk 2 above)
Comment (eyz) [reply pkg/sysctl/build.yml]: Perhaps we need an issue for changing the filesystem to readonly only after the image file adds from the YAML?

Comment (w9n) [reply pkg/sysctl/build.yml]: You can `mount --bind` files on a read only filesystem, this should work in your case aswell.

Comment (eyz) [reply pkg/sysctl/build.yml]: OK. I'll get more information and report back.

**Comment by w9n** [comment]
should not be necessary as successfully working in https://github.com/linuxkit/kubernetes/blob/master/yml/kube.yml#L53 and Bound in L13

**Comment by w9n** [comment]
everything going into the direction of https://www.freedesktop.org/software/systemd/man/sysctl.d.html as e.g. comments should be appreciated, i think.

**Comment by eyz** [comment]
@w9n  I'll update the code to support `;` as a comment prefix and `/` as a separator

**Comment by eyz** [comment]
The new comment and separator characters have been added, and `readonly: true' reverted back to before the PR

File: pkg/sysctl/main.go [hunk 3]
```diff
@@ -22,18 +22,21 @@ func init() {
...
-		return fmt.Errorf("Cannot parse %s", string(line))
+		return fmt.Errorf("Cannot parse %s", sysctlLineTrimmed)
 	}
 	// trim any extra whitespace
-	sysctlSetting, sysctlValue := strings.Trim(sysctlLineKV[0], " "), strings.Trim(sysctlLineKV[1], " ")
-	sysctlFile := filepath.Join(sysctlDir, filepath.Join(strings.Split(sysctlSetting, ".")...))
+	sysctlSetting, sysctlValue := strings.TrimSpace(sysctlLineKV[0]), strings.TrimSpace(sysctlLineKV[1])
+	sysctlFile := filepath.Join(sysctlDir, filepath.Join(strings.FieldsFunc(sysctlSetting, splitKv)...))
 	file, err := os.OpenFile(sysctlFile, os.O_WRONLY, 0)
+
```
Comment (justincormack) [inline pkg/sysctl/main.go]: can you remove this blank line, as the test below is for this `err` so a blank is a bit confusing.

**Comment by justincormack** [comment]
One very minor comment but otherwise LGTM

File: pkg/sysctl/main.go (hunk 3 above)
Comment (eyz) [reply pkg/sysctl/main.go]: Yes, agreed that was a bit odd. Thanks. Should be good now to proceed as of green CI.

**Comment by rn** [comment]
@eyz I've build and pushed the new package for x86 and arm64. Could you add a commit to update the YAMLs to the new hash, ie: `linuxkit/sysctl:4c1ef93bb5eb1a877318db4b2daa6768ed002e21`. That way the CI will pick up the new package

**Comment by eyz** [comment]
@rn I had to rebase against linuxkit/linuxkit upstream master. Once CI passes again, could you please rebuild the image and give me the new tag?

**Comment by rn** [comment]
The tag doesn't change from a rebase as it is based on the content hash. It's just the commit hash in the metadata/label which is out of date. I've force pushed an update

**Comment by eyz** [comment]
Updated image tag to linuxkit/sysctl:4c1ef93bb5eb1a877318db4b2daa6768ed002e21

**Comment by eyz** [comment]
@rn CI has completed, and is green
